app/Http/Controllers/ShopifyController.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers;
+
+use App\Http\Interfaces\DataProviderInterface;
+use GuzzleHttp\Client;
+use Psr\Http\Message\ResponseInterface;
+use \Symfony\Component\HttpFoundation\Response;
+
+class ShopifyController extends Controller
+{
+    public function getProducts(): array
+    {
+        $response = $this->fetchShopifyProducts();
```

Comment:
$products = $this->shopify->fetchProducts(); // return type should be wrapped by DTO, you can use Spatie/Data package for it

return $products->toArray();